File: cucumber/src/main/java/fr/polytech/models/Task.java
package fr.polytech.models;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Task {

    private Long id;
    private String name;
    private TaskType type;
    private TaskStatus status;
    private Date creationDate;
    private Date realizationDate;

    public Date getCreationDate() {
        return this.creationDate;
    }

    public void setCreationDate(Date creationDate) {
        this.creationDate = creationDate;
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public TaskType getType() {
        return this.type;
    }

    public void setType(TaskType type) {
        this.type = type;
    }

    public TaskStatus getStatus() {
        return this.status;
    }

    public void setStatus(TaskStatus status) {
        this.status = status;
    }


    public Date getRealizationDate() {
        return realizationDate;
    }

    public void setRealizationDate(Date realizationDate) {
        this.realizationDate = realizationDate;
    }
}

File: gatling/src/test/scala/models/Task.scala
package models

import java.util.Date

import models.TaskStatus.TaskStatus
import models.TaskType.TaskType

class Task {
  private var id = 0L
  private var name: String = null
  private var `type` : TaskType = null
  private var status : TaskStatus = null
  private var creationDate : Date = null
  private var realizationDate : Date = null

  def getCreationDate: Date = this.creationDate

  def creationDate(creationDate: Date) = {
    this.creationDate = creationDate : Date
  }

  def getId: Long = this.id

  def setId(id: Long): Unit = {
    this.id = id
  }

  def getName: String = this.name

  def setName(name: String): Unit = {
    this.name = name
  }

  def getType: TaskType = this.`type`

  def setType(`type`: TaskType): Unit = {
    this.`type` = `type`
  }

  def getStatus: TaskStatus = this.status

  def setStatus(status: TaskStatus): Unit = {
    this.status = status
  }


  def getRealizationDate: Date = realizationDate

  def setRealizationDate(realizationDate: Date): Unit = {
    this.realizationDate = realizationDate
  }
}
